### DISCUSSION TITLE
PHP 8.3 now available, Questions on which versions are usable with Laragon Full 6.0 220916

### DISCUSSION THREAD

**Discussion by Sheldon-Monk** [leokhoa/laragon#576] [Laragon's Articles & How-tos]
I see PHP 8.3 is now available.
I'm running Laragon Full 6.0 220916 on Win11 Pro (v.10.0.22621), with PHP 8.2.8-Win32-vs16-x64 [TS], and it's running fine; but the sites under development will be moved to Linux servers.  (Also running Apache httpd-2.4.57-win64-VS17 and MySQL-8.0.30-winx64)

On the php.net site, I see a number of download options, including what appears to be a separate one for Windows. [https://www.php.net/downloads.php](url).  Not sure which to use.  Also, on the ApacheLounge site [https://www.apachelounge.com/viewtopic.php?p=42093](url), there's commentary regarding the dropping of VS16 downloads:

> Original Subject: Drop VS16 downloads :: OpenSSL 1.1.1 End of Life 2023-09-11

Update:

On 24 July You can now download VS17 Win32 at https://www.apachelounge.com/viewtopic.php?p=42099

On 31 May we dropped the VS16 builds and moved the last VS16 to the Additonal download page.

Did not realized that are quite some (professionally)users are missing Win32 now.
See below for a plan to make Win32 available for VS17.

https://www.openssl.org/blog/blog/2023/03/28/1.1.1-EOL/

It is advised to upgrade to VS17, all VS16 binaries and modules are compatible with VS17.

Note on PHP: Only PHP 8.2 and 8.1 (build with 3.0.x) is running as module.
It is strongly advised to run e.g. PHP with mod_fcgid.

VS17 is now only Win64.
VS16 Win32 is hardly downloaded.

Do you need still a VS17 Win32 ?

Please comment.

Last edited by Steffen on Mon 24 Jul '23 17:05; edited 19 times in total >

**Could use some suggestions which to use; the steps to do so (in detail please)**

**Comment by Pen-y-Fan** [comment]
PHP 8.3 isn't available to download for Windows, yet. RC1 is available for Linux. When the Windows version is released it can be downloaded from [Windows PHP.net download](https://windows.php.net/download/).

For details on which VS runtime you need see my blog [How to update the PHP version in Laragon](https://pen-y-fan.github.io/2023/01/15/how-to-update-the-php-version-in-laragon/)

**Reply by Sheldon-Monk** [reply]
Okay.  What about the other issues mentioned in the Update?

**Reply by Sheldon-Monk** [reply]
Such as:  Can you comment on the issues of VS 17 Win32 and the VS16 version; also, how does the x64 version affect this.  Some of us are not as knowledgeable as others, and could use the benefit of input (as compared to spending hours upon hours in trial-and-error.  Thanks.

**Reply by Pen-y-Fan** [reply]
> Such as: Can you comment on the issues of VS 17 Win32 and the VS16 version; also, how does the x64 version affect this. Some of us are not as knowledgeable as others, and could use the benefit of input (as compared to spending hours upon hours in trial-and-error. Thanks.

PHP 8.3 isn't released, yet, so we don't know what VS version will be used. If it is bumped from VS16 to VS17, the runtime will be free to download and install from Microsoft, there is nothing to be concerned with. Win32 will still be required if we run old legacy PHP builds, for new builds 64-bit is the standard.

For details on which VS, 32 bit vs 64 bit and thread safety see my blog [How to update the PHP version in Laragon](https://pen-y-fan.github.io/2023/01/15/how-to-update-the-php-version-in-laragon/)

**Reply by Sheldon-Monk** [reply]
Thanks!  Very Helpful and Appreciated!

**Reply by NarayananVT** [reply]
When we can expect php_sqlsrv dll for php 8.3

**Reply by HenkPoley** [reply]
@NarayananVT they won't know at Laragon. They don't make Microsoft SQL Server. You can find info about php_sqldrv for PHP 8.3 here: https://github.com/microsoft/msphpsql/issues/1477

**Comment by Avori1** [comment]
![2024-09-12_20-25-33](https://github.com/user-attachments/assets/ad1b052a-8f86-4f5c-bd07-b01b2fa7a3d9)
Here is the issue I'm getting. That's me just downloading either 8.3 or 8.2 from https://windows.php.net/download/ and adding it to Laragon. Any ideas on how to fix it? 

Also, seems like the XML extension doesn't come with php for windows by default, anyone have a fix for that?

**Reply by HenkPoley** [reply]
You probably need a newer version of Apache as well.

**Reply by Avori1** [reply]
Yep, that was the case. Found it in another thread. Thanks!

**Comment by Sheldon-Monk** [comment]
I am attempting to upgrade Apache and PHP, to httpd-2.4.62-240904-win64-vs17 & 8.3.4-Win32-vs16-x64 respectively. The PHP update worked, but the Apache generated an error (see attachment).
Two problems:
(1) Trying to figure out why the error; but more importantly, problem (2)
(2) The error message dialogue box is not expandable and the error message inside of it needs more room to be read and understood.

Also, I don't understand why the Laragon program dialogue box itself is also not expandable,  Why???

(I am updating from PHP 8.1.10-Win32-va16-x64 and Apache httpd-2.4.59-240404-win64-vs17; both of which worked correctly.)

Any help appreciated.

![Screenshot 2024-09-13 090206](https://github.com/user-attachments/assets/6e8b640d-3546-41a0-89c0-a52986890e3f)

**Comment by Avori1** [comment]
Hey, I had this same issue, and it’s an easy fix. The folder you downloaded, httpd-2.4.62-240904-win64-VS17, contains two files and another folder named Apache24.

To resolve the issue, you need to move the contents of the Apache24 folder up one level. Here’s how:

Open the Apache24 folder inside httpd-2.4.62-240904-win64-VS17.
Select all the files and folders inside Apache24.
Move them to the httpd-2.4.62-240904-win64-VS17 folder, so they are no longer inside Apache24.
In your Laragon Apache directory, you should now have a folder named httpd-2.4.62-240904-win64-VS17 that directly contains the bin folder and other files, not another directory inside it.

If you look at the old Apache version folder in Laragon, you’ll know what I mean, see that the bin folder and other files are directly inside the main version folder, not nested within another directory.

Hope that makes it clearer.

**Comment by Sheldon-Monk** [comment]
THANK YOU, THAT WORKED, (.... SORT OF)!

It solved the Apache update issue, but is now producing a threat notification in Malwarebytes, and that app is quarantining \laragon\bin\apache\httpd-2.4.62-240904-win64-VS17\bin\httpd.exe file.

The result is that the quarantining deactivates the use of the file.

Any ideas???

Thanks again.


![Malwarebytes Ransomeware Blocked in Latagon-bin-apache  after new v install - 09142024](https://github.com/user-attachments/assets/5a60db31-d308-4e68-a54e-91eab9dc053c)
![Malwarebytes Quarantine folder - 09142024](https://github.com/user-attachments/assets/7636bd2d-2d01-4c72-b51b-6ea08c114f65)
![Malwarebytes - Explanation of MRAG - 09142024](https://github.com/user-attachments/assets/d39cecb5-7531-49d6-8e70-73202ff8687b)

**Reply by Avori1** [reply]
Did you download apache from their official website? If so, see if you can make an exception for that file. Otherwise go download it from the official website.

**Comment by AufarZakky** [comment]
![Screenshot 2025-03-06 151905](https://github.com/user-attachments/assets/b65f2674-49a0-4528-9116-d9fc2cb81574)
can u help me about this bro..
how to configure oci8 on laravel
